app.rb

```diff
@@ -0,0 +1,99 @@
+require_relative 'book'
+require_relative 'teacher'
+require_relative 'student'
+require_relative 'rental'
+require_relative 'person'
+
+class App
+  attr_accessor :books, :persons, :rentals
+
+  def initialize
+    @books = []
+    @persons = []
+    @rentals = []
+  end
+
+  # list all books
+  def list_books
+    puts 'List of books'
+    @books.each { |book| puts "title : #{book.title} by author : #{book.author}" }
+  end
+
+  def list_persons
+    @persons.each do |person|
+      if person.instance_of?(Teacher)
+        puts "[Teacher]Age: #{person.age} Name: #{person.name}, ID: #{person.id}"
+      else
+        puts "[Student]Age: #{person.age},Name: #{person.name}, ID: #{person.id} "
+      end
+    end
+  end
```

Comment:
- [ ] Also here, although I am unable to create a teacher, however, when listing the people, I noticed that the student's name is not displayed as intended during its creation ⚠ Please ensure that this error is also addressed for teachers. Since this issue was not present during the last review, I'm marking it as required. Please rectify this.

Last review | Current implementation 
---| ---|
![image](https://github.com/supreeti/OPPS-Library-Management-System/assets/106119105/2c907f6e-ee25-464b-8252-02af6f67969b) | ![image](https://github.com/supreeti/OPPS-Library-Management-System/assets/106119105/5d0f95be-fca9-49f9-9859-b4d077e84aca)

Reply:
Fixed.